src/dapla_pseudo/v1/mutable_dataframe.py

```diff
@@ -45,55 +45,92 @@ def __init__(
         self.func = func
         self.target_func = target_func
 
-    def update_col(self, key: str, data: list[str]) -> None:
-        """Update the values in the matched column."""
-        self.col.update({key: data})
+    def get_value(self) -> list[str | int | None]:
+        """Get the inner value.
+
+        If hierarchical, get the values of the matched column.
```

Comment:
Would be nice if we could deduce this somehow?

Maybe document what how this affects the processing